src/prefect/engine/cloud/flow_runner.py

```diff
@@ -145,8 +148,7 @@ def call_runner_target_handlers(self, old_state: State, new_state: State) -> Sta
             state.state = old_state  # type: ignore
             raise ENDRUN(state=state)
 
-        if version is not None:
-            prefect.context.update(flow_run_version=version + 1)
+        prefect.context.update(flow_run_version=(version or 0) + 1)
```

Comment:
It would be really unusual but do you want to add the same logic [here](https://github.com/PrefectHQ/prefect/blob/version_locking/src/prefect/engine/cloud/task_runner.py#L156)?